src/shapes/itext.class.js

```diff
@@ -636,12 +636,12 @@
           offset = this._fontSizeFraction * lineHeight / this.lineHeight;
 
       if (decl) {
-        var shouldStroke = decl.stroke || this.stroke,
-            shouldFill = decl.fill || this.fill;
+        var shouldStroke = (decl.stroke || this.stroke) && method === 'strokeText',
+            shouldFill = (decl.fill || this.fill) && method === 'fillText';
```

Comment:
We where both stroking and filling when calling method = 'strokeText' or 'fillText'